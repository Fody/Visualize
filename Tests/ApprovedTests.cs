using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;

[UseReporter(typeof(DiffReporter))]
#if(DEBUG)
[UseApprovalSubdirectory("results-debug")]
#else
[UseApprovalSubdirectory("results-release")]
#endif
[TestFixture]
public class ApprovedTests
{
    [Test]
    public void ClassWithProperties()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithProperties"));
    }

    [Test]
    public void ClassWithConst()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithConst"));
    }

    [Test]
    public void ClassWithIndexor()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIndexor"));
    }

    [Test]
    public void ClassWithDataAnnotations()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithDataAnnotations"));
    }

    [Test]
    public void ClassWithIEnumerable()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIEnumerable"));
    }

    [Test]
    public void ClassWithICollection()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithICollection"));
    }

    [Test]
    public void GenericClassWithIEnumerable()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.GenericClassWithIEnumerable`1"));
    }

    [Test]
    public void GenericClassWithICollection()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.GenericClassWithICollection`1"));
    }

    [Test]
    public void ClassWithIEnumerableNotAttributed()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIEnumerableNotAttributed"));
    }

    //[Test]
    //public void Sample()
    //{
    //    var sampleClassType = AssemblyWeaver.Assembly.GetType("AssemblyToProcess.ClassWithIEnumerable");
    //    var sample = Activator.CreateInstance(sampleClassType);
    //}
}