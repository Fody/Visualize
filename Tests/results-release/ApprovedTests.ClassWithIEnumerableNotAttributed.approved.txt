.class public auto ansi beforefieldinit AssemblyToProcess.ClassWithIEnumerableNotAttributed
       extends [mscorlib]System.Object
       implements class [mscorlib]System.Collections.Generic.IEnumerable`1<int32>,
                  [mscorlib]System.Collections.IEnumerable
{
  .method public hidebysig newslot virtual final 
          instance class [mscorlib]System.Collections.Generic.IEnumerator`1<int32> 
          GetEnumerator() cil managed
  {
    // Code size       14 (0xe)
    .maxstack  8
    .language '{3F5162F8-07C6-11D3-9053-00C04FA302A1}', '{994B45C4-E6E9-11D2-903F-00C04FA302A1}', '{5A869D0B-6611-11D3-BD2A-0000F80849BD}'
    .line 11,11 : 13,60 'AssemblyToProcess\\ClassWithIEnumerableNotAttributed.cs'
    IL_0000:  ldc.i4.0
    IL_0001:  ldc.i4.s   10
    IL_0003:  call       class [mscorlib]System.Collections.Generic.IEnumerable`1<int32> [System.Core]System.Linq.Enumerable::Range(int32,
                                                                                                                                    int32)
    IL_0008:  callvirt   instance class [mscorlib]System.Collections.Generic.IEnumerator`1<!0> class [mscorlib]System.Collections.Generic.IEnumerable`1<int32>::GetEnumerator()
    IL_000d:  ret
  } // end of method ClassWithIEnumerableNotAttributed::GetEnumerator
  .method private hidebysig newslot virtual final 
          instance class [mscorlib]System.Collections.IEnumerator 
          System.Collections.IEnumerable.GetEnumerator() cil managed
  {
    .override [mscorlib]System.Collections.IEnumerable::GetEnumerator
    // Code size       7 (0x7)
    .maxstack  8
    .line 16,16 : 13,36 ''
    IL_0000:  ldarg.0
    IL_0001:  call       instance class [mscorlib]System.Collections.Generic.IEnumerator`1<int32> AssemblyToProcess.ClassWithIEnumerableNotAttributed::GetEnumerator()
    IL_0006:  ret
  } // end of method ClassWithIEnumerableNotAttributed::System.Collections.IEnumerable.GetEnumerator
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
    IL_0006:  ret
  } // end of method ClassWithIEnumerableNotAttributed::.ctor
} // end of class AssemblyToProcess.ClassWithIEnumerableNotAttributed